cmd/queue/main.go

```diff
@@ -264,7 +267,7 @@ func buildServer(ctx context.Context, env config, probeContainer func() bool, st
 	}
 	composedHandler = queue.ProxyHandler(breaker, stats, tracingEnabled, composedHandler)
 	composedHandler = queue.ForwardedShimHandler(composedHandler)
-	composedHandler = handler.NewTimeoutHandler(composedHandler, "request timeout", firstByteTimeout, idleTimeout)
+	composedHandler = handler.NewTimeoutHandler(composedHandler, "request timeout", firstByteTimeout, idleTimeout, maxDurationTimeout)
```

Comment:
The implementation of this timeout handler is inspired by the one in std lib as also stated [here](https://github.com/knative/serving/blob/a794e1fdcbd6652d81af93a430afe94fdcf9abb0/pkg/http/handler/timeout.go#L55).
Since we are doing a custom handler for first byte and there is another PR for idle time out coing it makes sense to have all the logic in one place.

Reply:
ok